### PR TITLE
WT-8598 Avoid checkpoint cleanup always on shutdown

### PR DESCRIPTION
**Pull request by kommiharibabu**
Earlier checkpoint cleanup was avoided on shutdown only when
the timestamps in use. Checkpoint cleanup is a time consuming
operation, so avoid it always on shutdown irrespective of timestamps.

### CODE REVIEW

File: src/txn/txn_ckpt.c [hunk 1]
```diff
@@ -352,11 +352,8 @@ __checkpoint_reduce_dirty_cache(WT_SESSION_IMPL *session)
     conn = S2C(session);
     cache = conn->cache;
 
-    /*
-     * Give up if scrubbing is disabled, including when checkpointing with a timestamp on close (we
-     * can't evict dirty pages in that case, so scrubbing cannot help).
-     */
-    if (F_ISSET(conn, WT_CONN_CLOSING_TIMESTAMP) || cache->eviction_checkpoint_target < DBL_EPSILON)
+    /* Give up if scrubbing is disabled. */
```
Comment (kommiharibabu) [inline src/txn/txn_ckpt.c:355]: With WT-8379, eviction is allowed in parallel to the shutdown checkpoint. The comment and code don't valid now, so I removed them to allow scrubbing in the shutdown checkpoint. Let me know if I can move these changes into a separate ticket?

**Review by keithbostic — APPROVED** [review]
lgtm

**Review by sueloverso — APPROVED** [review]
LGTM. My only comment is that the only real remaining use of WT_CONN_CLOSING_TIMESTAMP is in conn_dhandle.c. The only other one is a couple lines down in txn_ckpt.c which could easily use a local boolean. Is there any way to eliminate that last one with something like WT_CONN_CLOSING_CHECKPOINT && <something else>?? If it is too risky to get right, then ignore this suggestion.

**Comment by kommiharibabu** [comment]
@sueloverso Thanks for the comment. I tried that approach in the first commit and later reverted it. Upon further checking, I don't see a need for that check in the conn_dhandle.c as no other thread closes the dhandle while we perform the final checkpoint. So I removed that check completely. Am I missing anything?

**Comment by kommiharibabu** [comment]
@sueloverso I found the reason behind the need for the check. It can be changed with a stable timestamp and CLOSING_CHECKPOINT check. I will run the stress tests on this PR. If everything is good, I will merge it.

**Comment by kommiharibabu** [comment]
All stress test failures are not related to the PR and there exists a similar problem in the develop also that failed with the same call stack.

**Comment by sueloverso** [comment]
@kommiharibabu I think the difference is that the old flag indicated that we were closing, checkpointing and using the stable timestamp (i.e. use_timestamp=true). MongoDB will be running with a stable timestamp set, but calling close with use_timestamp=false explicitly. So a check for the general checkpoint flag and a stable timestamp set be true and that is a new code path that was previously not true.

**Comment by kommiharibabu** [comment]
@sueloverso Thanks for the details. Yes, you are correct that the old flag is not set when the use_timestamp = false. Earlier with use_timestamp = true, checkpoint writes only some part of the modified data, now with durable history, it writes everything that is committed even with use_timestamp=true. So there will not be anything that is left out to be discarded. 

Do you see any problem with the added approach? Maybe I can revert the change and test the combinations?

**Pull request merged**